src/rpdk/core/project.py

```diff
@@ -1345,21 +1345,91 @@ def _generate_stack_template_files(self) -> None:
                 json_data = json.load(f)
             resource_name = self.type_info[2]
-            stack_template_data = {
-                "Description": f"Template for {self.type_name}",
-                "Resources": {
-                    f"{resource_name}": {
-                        "Type": self.type_name,
-                        "Properties": self._replace_dynamic_values(
-                            json_data["CreateInputs"]
-                        ),
-                    }
-                },
-            }
-            stack_template_file_name = f"{CANARY_FILE_PREFIX}{count}_001.yaml"
-            stack_template_file_path = stack_template_folder / stack_template_file_name
-            with stack_template_file_path.open("w") as stack_template_file:
-                yaml.dump(stack_template_data, stack_template_file, indent=2)
+
+            self._save_stack_template_data(
+                resource_name,
+                count,
+                stack_template_folder,
+                self._replace_dynamic_values(
+                    json_data["CreateInputs"],
+                ),
+                "001",
+            )
+            if "PatchInputs" in json_data:
+                deepcopy_create_input = copy.deepcopy(json_data["CreateInputs"])
+                self._save_stack_template_data(
+                    resource_name,
+                    count,
+                    stack_template_folder,
+                    self._apply_patch_inputs_to_create_inputs(
+                        json_data["PatchInputs"],
+                        deepcopy_create_input,
+                    ),
+                    "002",
+                )
+
+    def _save_stack_template_data(
+        self,
+        resource_name,
+        contract_test_input_count,
+        stack_template_folder,
+        properties_data,
+        suffix,
+    ):
+        stack_template_data = {
+            "Description": f"Template for {self.type_name}",
+            "Resources": {
+                f"{resource_name}": {
+                    "Type": self.type_name,
+                    "Properties": properties_data,
+                }
+            },
+        }
+        stack_template_file_name = (
+            f"{CANARY_FILE_PREFIX}{contract_test_input_count}_{suffix}.yaml"
+        )
+        stack_template_file_path = stack_template_folder / stack_template_file_name
+        with stack_template_file_path.open("w") as stack_template_file:
+            yaml.dump(stack_template_data, stack_template_file, indent=2)
+
+    def _apply_patch_inputs_to_create_inputs(
+        self, patch_inputs: Dict[str, Any], create_inputs: Dict[str, Any]
+    ) -> Dict[str, Any]:
+        for patch_input in patch_inputs:
+            self._apply_patch_input_to_create_input(patch_input, create_inputs)
+        return create_inputs
+
+    def _apply_patch_input_to_create_input(
+        self, patch_input: Any, create_inputs: Dict[str, Any]
+    ) -> Dict[str, Any]:
+        op = patch_input.get("op")
```

Comment:
Instead of manually handling patch operation was any investigation done into using existing tools for this? For example `jsonpatch`?

Reply:
i considered using `pydash`. i think `jsonpatch` would be best here